{
  "id": "oval:com.redhat.cve:def:20213281",
  "version": "636",
  "class": "vulnerability",
  "metadata": {
    "title": "CVE-2021-3281 django: Potential directory-traversal via archive.extract() (moderate)",
    "reference": [
      {
        "ref_id": "CVE-2021-3281",
        "ref_url": "https://access.redhat.com/security/cve/CVE-2021-3281",
        "source": "CVE"
      }
    ],
    "description": "DOCUMENTATION: A flaw was found in django where the`django.utils.archive.extract()` function, used by `startapp --template` and `startproject --template`, allowed directory-traversal via an archive with absolute paths or relative paths with dot segments. \n            STATEMENT: The following products ship affected version of python-django, however the vulnerable function archive.extract() is currently not used in any part of the product and hence this issue has been rated as having a security impact of Low:\n* Red Hat Gluster Storage 3\n* Red Hat Update Infrastructure 3\n\nBecause the flaw's impact is lower and Red Hat OpenStack Platform 13 will be retiring soon, no update will be provided at this time for the RHOSP13 python-django package.",
    "advisory": {
      "from": "secalert@redhat.com",
      "severity": "Moderate",
      "updated": {
        "date": "2022-04-17"
      },
      "cve": [
        {
          "text": "CVE-2021-3281",
          "cvss3": "5.3/CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:L/A:N",
          "cwe": "CWE-22",
          "href": "https://access.redhat.com/security/cve/CVE-2021-3281",
          "impact": "moderate",
          "public": "20210201:1000"
        }
      ],
      "affected": {
        "resolution": [
          {
            "state": "Will not fix",
            "component": [
              "python-django",
              "python-django-bash-completion"
            ]
          }
        ]
      },
      "affected_cpe_list": {
        "cpe": [
          "cpe:/a:redhat:ceph_storage:3",
          "cpe:/a:redhat:ceph_storage:3::el7"
        ]
      },
      "issued": {}
    },
    "affected": {}
  },
  "criteria": {
    "operator": "OR",
    "criterias": [
      {
        "operator": "AND",
        "criterias": [
          {
            "operator": "OR",
            "criterias": [
              {
                "operator": "AND",
                "criterions": [
                  {
                    "test_ref": "oval:com.redhat.cve:tst:201814574001",
                    "comment": "python-django-bash-completion is installed"
                  },
                  {
                    "test_ref": "oval:com.redhat.cve:tst:201814574002",
                    "comment": "python-django-bash-completion is signed with Red Hat redhatrelease2 key"
                  }
                ]
              },
              {
                "operator": "AND",
                "criterions": [
                  {
                    "test_ref": "oval:com.redhat.cve:tst:201814574003",
                    "comment": "python-django is installed"
                  },
                  {
                    "test_ref": "oval:com.redhat.cve:tst:201814574004",
                    "comment": "python-django is signed with Red Hat redhatrelease2 key"
                  }
                ]
              }
            ]
          }
        ],
        "criterions": [
          {
            "test_ref": "oval:com.redhat.cve:tst:201610745003",
            "comment": "Red Hat Enterprise Linux 7 is installed"
          }
        ]
      }
    ],
    "criterions": [
      {
        "test_ref": "oval:com.redhat.cve:tst:201610745004",
        "comment": "Red Hat Enterprise Linux must be installed"
      }
    ]
  }
}
